howso/details_residuals.amlg

```diff
@@ -30,30 +30,53 @@
 		)
 
 		(declare (assoc
-			local_feature_residuals
+			local_feature_metrics
 				(if (>= model_size 2)
-					(get
-						(call !ExpandResidualValuesToUncertainty  (assoc
-							feature_residuals_map
-								(call !CalculateFeatureResiduals (assoc
-									;only the features from the details
-									features relevant_features
-									case_ids (indices regional_model_cases_map)
-									regional_model_only (true)
-									robust_residuals robust_residuals
-									;use the same hyperparameters that were used to make the original prediction
-									custom_hyperparam_map hyperparam_map
-								))
-						))
-						"residual_map"
+					(call !ExpandResidualValuesToUncertainty  (assoc
+						feature_residuals_map
+							(call !CalculateFeatureResiduals (assoc
+								;only the features from the details
+								features relevant_features
+								case_ids (indices regional_model_cases_map)
+								regional_model_only (true)
+								robust_residuals robust_residuals
+								compute_all_statistics compute_all_statistics
+								store_values store_values
+								;use the same hyperparameters that were used to make the original prediction
+								custom_hyperparam_map hyperparam_map
+							))
+					))
+
+					;can't compute residuals or prediction stats if model is too small
+					(assoc
+						"residual_map" (zip relevant_features 0)
+						"prediction_stats" (zip relevant_features (assoc))
 					)
-					;can't compute residuals if model is too small
-					(zip relevant_features 0)
 				)
 		))
 
-		;append feature deviations to output
-		(accum (assoc output (assoc "feature_residuals" local_feature_residuals) ))
+		;append feature deviations to output if selected
+		(if (get details "feature_residuals")
+			(accum (assoc output (assoc "feature_residuals" (get local_feature_metrics "residual_map"))))
+		)
+
+		;append prediction stats to output if selected
+		(if (get details "prediction_stats")
+			(accum (assoc output (assoc 
+				"prediction_stats" 
+				(append 
+					(remove 
+						(get local_feature_metrics "prediction_stats")
+						;Keep the confusion_matrix if it is selected for
+						(if (get details "confusion_matrix")
+							(list ".hyperparam_path" ".robust")
+							(list ".hyperparam_path" ".robust" "confusion_matrix")
+						)
+					)
+					(assoc "mae" (get local_feature_metrics "residual_map"))
```

Comment:
keys in (assoc)'s don't need double quotes. This should be fine, but something to know.